internal/tofu/eval_context_builtin.go

```diff
@@ -145,6 +147,18 @@ func (ctx *BuiltinEvalContext) InitProvider(addr addrs.AbsProviderConfig) (provi
 		return nil, err
 	}
 
+	if ctx.Config != nil {
```

Comment:
The config is already accessed in EvaluationScope below:
https://github.com/opentofu/opentofu/pull/1772/files?diff=split&w=1#diff-126d6884fd43891cbb56845ee6f7f51ed02895af69190815e5c694a51a3777d3L522

I don't think we need to add ctx.Config and can use that same code instead.

Reply:
Nice catch, thanks!

[5fa601a](https://github.com/opentofu/opentofu/pull/1772/commits/5fa601a8bae2f435f88d08e2b390d95febabbb6d)